lib/soft_deletion/core.rb

```diff
@@ -94,12 +94,24 @@ def soft_delete_dependencies
 
     protected
 
+    def soft_delete_counter_cache_associations
+      each_counter_cached_associations do |association|
+        foreign_key = association.reflection.foreign_key.to_sym
+        unless destroyed_by_association && destroyed_by_association.foreign_key.to_sym == foreign_key
```

Comment:
inlining the `foreign_key` would save the extra work to get the variable in the case it was not destroyed by an association ...

Reply:
like `destroyed_by_association.foreign_key.to_sym == association.reflection.foreign_key.to_sym`